test/apps/iodemo/io_demo.cc

```diff
@@ -2028,7 +2106,17 @@ class DemoClient : public P2pDemoCommon {
     void active_servers_make_unused(size_t active_index) {
         assert(active_index < _num_active_servers_to_use);
         --_num_active_servers_to_use;
+        // Swap the removed element and the last element to use
         active_servers_swap(active_index, _num_active_servers_to_use);
+
+        if (active_index < _next_active_index) {
```

Comment:
@evgeny-leksikov the all solutions we discussed are not suitable actually to fix two things:
- IO-demo must visit all connections which are not visited yet, i.e. their index >= `_next_active_index`
- `_next_active_index` should become `0` if it was equal to `_num_active_servers_to_use` after doing `--_num_active_servers_to_use`.

So, I can't find suitable solution without decrementing `_next_active_index` if `active_index < _next_active_index`, because the last active server moves after `swap()` to the part of `_active_servers` where visited server are placed.